decepticonlp/attacks/attack.py

```diff
@@ -0,0 +1,328 @@
+from tqdm import tqdm
+import torch
+
+loss_type_list = [
+    torch.nn.modules.loss.BCELoss,
+    torch.nn.modules.loss.BCEWithLogitsLoss,
+    torch.nn.modules.loss.CrossEntropyLoss,
+    torch.nn.modules.loss.NLLLoss,
+]
+
+
+class CharAttackerWithHuggingFace:
+    """
+		Args:
+		model: PyTorch model
+			-Pretrained PyTorch model (trained on benign training set) for inference
+		
+		data_loader: torch.utils.data.Dataloader 
+			-Dataloader object of the original test set
+		
+		adversarial_data_loader: torch.utils.data.Dataloader
+			-Dataloader object of the adversarial test set
+		
+		input_format: list (of strings) (default: ['input_ids','labels'])
+			-The input format as you would feed to your model.
+			Eg:
+				For BERTForSequenceClassification, the input_format will be ['input_ids','attention_masks','labels'] (if you are feeding only these three to the model).
+		
+		accuracy: boolean (default: True)
+			-If True, logs and prints the accuracy
+		
+		logs_after_every: int (default: 50)
+			-number of batch intervals after which loss (and accuracy) is/are logged.
+		
+		device: torch.device (default: torch.device('cpu'))
+			-to push the inputs/model to CPU/GPU.
+	"""
+
+    def __init__(
+        self,
+        model,
+        data_loader,
+        adversarial_data_loader,
+        input_format,
+        accuracy=True,
+        logs_after_every=50,
+        device=torch.device("cpu"),
+    ):
+
+        # initialise parameters
+        self.model = model.to(device)
+        self.data_loader = data_loader
+        self.adversarial_data_loader = adversarial_data_loader
```

Comment:
Why do we need to have 2 data loaders? 

cc: @rajaswa

Reply:
One for the original dataset, and the other for the adversarial dataset (the user will apply transforms). I discussed this with Rajaswa and we thought that instead of passing dataloaders, we could pass the Dataset objects. 

However, during implementation, I came across multiple examples online where users have used torchtext iterators, and passing them through these classes would also work, as they are handled the exact same way as dataloaders. So, it comes down to flexibility, which is why I chose passing dataloaders over passing Dataset objects.